### DISCUSSION TITLE
Graphics API Considerations

### DISCUSSION THREAD

**Discussion by Syncaidius** [Syncaidius/MoltenEngine#92] [General]
I'm currently looking into whether it is worth keeping plans to support OpenGL, or whether it's better to drop OpenGL in favor of starting earlier on Vulkan support. A few aspects being taken into consideration are:
- What percentage of active devices support Vulkan? (All form-factors)
- How many devices would be left out by the lack of OpenGL support?
- How many non-vulkan devices will be left in target market(s) by the time OpenGL support is completed?
- Would OpenGL support allow DirectX 11 support to be dropped in order to reduce code overhead/maintainance?
- Do we need DirectX 11 or can it be replaced by OpenGL?
- How much code can be shared between DirectX 12 and Vulkan, if both were supported?
- Is vulkan support on Android high enough to drop OpenGL? https://developer.android.com/about/dashboards#Vulkan

Any thoughts on this would be greatly appreciated!

**Comment by IceReaper** [comment]
OpenGL ends with 4.1 on MacOS. Vulkan works through moltenvk.
DirectX < OpenGL < Vulkan
Way more devices Support OpenGL then directx.
But Vulkan ist pretty widely supported nowadays.
My Vote would be to Go full Vulkan directly. No Microsoft only API, No old gl 4.1 API, No gl 4.6 API which Kicks MacOS Out of compatibility.
